corehq/apps/enterprise/models.py

```diff
@@ -53,11 +64,29 @@ def get_domains(cls, source_domain):
             return []
         return list(set(config.domains) - {config.source_domain})
 
+    @classmethod
+    @quickcache(['source_domain'], timeout=7 * 24 * 60 * 60)
+    def is_source_domain(cls, source_domain):
+        """
+        Returns true if given domain is the source domain for an enabled configuration.
+        """
+        try:
+            cls.objects.get(is_enabled=True, source_domain=source_domain)
+        except cls.DoesNotExist:
+            return False
+        return True
```

Comment:
nit: naming the parameter `source_domain` is _slightly_ confusing since it might not be.

```suggestion
    @quickcache(['domain'], timeout=7 * 24 * 60 * 60)
    def is_source_domain(cls, domain):
        """
        Returns true if given domain is the source domain for an enabled configuration.
        """
        try:
            cls.objects.get(is_enabled=True, source_domain=domain)
        except cls.DoesNotExist:
            return False
        return True
```

Reply:
makes sense - 7b63c702c15